src/Umbraco.Web.Common/Middleware/BootFailedMiddleware.cs

```diff
@@ -65,14 +66,20 @@ public async Task InvokeAsync(HttpContext context, RequestDelegate next)
         }
     }
 
-    private string GetBootErrorFileName()
+    private IFileInfo? GetBootErrorFileInfo()
     {
-        var fileName = _webHostEnvironment.MapPathWebRoot("~/config/errors/BootFailed.html");
-        if (File.Exists(fileName))
+        IFileInfo? fileInfo = _webHostEnvironment.WebRootFileProvider.GetFileInfo("config/errors/BootFailed.html");
+        if (fileInfo.Exists)
+        {
+            return fileInfo;
+        }
+
+        fileInfo = _webHostEnvironment.WebRootFileProvider.GetFileInfo("umbraco/views/errors/BootFailed.html");
```

Comment:
Should the static `umbraco/` part be replaced with the customized `Umbraco:CMS:Global:UmbracoPath` setting? This probably needs issue https://github.com/umbraco/Umbraco-CMS/issues/12593 to be fixed first though.